### PR TITLE
Associate Organization and Project in Quarterdeck

### PR DESCRIPTION
**Pull request by bbengfort**
### Scope of changes

Adds an endpoint for Tenant to associate an organization and a project in Quarterdeck so that Quarterdeck can securely issue APIKeys for a project, verifying that the project belongs to the Organization in the claims of the requester.

Fixes SC-13077

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Is there anything Tenant needs other than what's here? 

Are these utilities usable for Tenant and does it close the security hole?

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.

### CODE REVIEW

File: pkg/quarterdeck/quarterdeck.go [hunk 1]
```diff
@@ -261,11 +262,17 @@ func (s *Server) setupRoutes() (err error) {
...
-			apikeys.GET("", middleware.Authorize("apikeys:read"), s.APIKeyList)
-			apikeys.POST("", middleware.Authorize("apikeys:edit"), s.APIKeyCreate)
-			apikeys.GET("/:id", middleware.Authorize("apikeys:read"), s.APIKeyDetail)
-			apikeys.PUT("/:id", middleware.Authorize("apikeys:edit"), s.APIKeyUpdate)
-			apikeys.DELETE("/:id", middleware.Authorize("apikeys:delete"), s.APIKeyDelete)
+			apikeys.GET("", middleware.Authorize(perms.ReadAPIKeys), s.APIKeyList)
+			apikeys.POST("", middleware.Authorize(perms.EditAPIKeys), s.APIKeyCreate)
+			apikeys.GET("/:id", middleware.Authorize(perms.ReadAPIKeys), s.APIKeyDetail)
+			apikeys.PUT("/:id", middleware.Authorize(perms.EditAPIKeys), s.APIKeyUpdate)
+			apikeys.DELETE("/:id", middleware.Authorize(perms.DeleteAPIKeys), s.APIKeyDelete)
```
Comment (pdeziel) [inline pkg/quarterdeck/quarterdeck.go:269]: Thanks for updating these!

File: pkg/quarterdeck/tokens/claims.go [hunk 2]
```diff
@@ -31,3 +35,27 @@ func (c Claims) HasAllPermissions(requiredPermissions ...string) bool {
...
+// themselves; otherwise the Null ULID will prevent most accesses from succeeding.
+func (c Claims) ParseOrgID() ulid.ULID {
+	orgID, err := ulid.Parse(c.OrgID)
+	if err != nil {
+		return ulids.Null
+	}
+	return orgID
+}
+
+// ParseUserID reteurns the ULID of the user from the Subject of the claims. If the
```
Comment (pdeziel) [inline pkg/quarterdeck/tokens/claims.go:51]:
```suggestion
// ParseUserID returns the ULID of the user from the Subject of the claims. If the
```

**Review by pdeziel — APPROVED** [review]
I think this fixes the gap, thanks for adding the helpers and the extra endpoint. Do we need to create a story on the tenant side to call `ProjectCreate` when new tenant projects are created?

**Comment by bbengfort** [comment]
@pdeziel created SC-13122 to follow on from this story for tenant integration.

**Pull request merged**